### PR TITLE
Fix Activity.ParentId trace flags part

### LINKED ISSUE #48781: Activity ParentId Does Not Format flags
### Description
When using the SetParentId overload that takes in the individual traceId, spanId and flags the ActivityTraceFlags get set correctly. However when calling the lazy ParentId getter the format is hard coded to return no flags ("-00"). Why is this format logic ignoring the ActivityTraceFlags when the Id getter does do the correct formatting? 

https://github.com/dotnet/runtime/blob/970d347a1b06951692cfecc1cc12a500158708b1/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L539

https://github.com/dotnet/runtime/blob/970d347a1b06951692cfecc1cc12a500158708b1/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L223

https://github.com/dotnet/runtime/blob/970d347a1b06951692cfecc1cc12a500158708b1/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L197

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes #48781

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]

**Comment by tarekgh** [comment]
Thanks @stephentoub 

@noahfalk please let me know if you have any comment on this one before I merge it.

File: src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs [hunk 1]
```diff
@@ -1694,6 +1694,21 @@ public void StructEnumerator_TagsLinkedList()
...
+            a.SetIdFormat(ActivityIdFormat.W3C);
+            a.SetParentId(ActivityTraceId.CreateFromString("0123456789abcdef0123456789abcdef".AsSpan()), ActivitySpanId.CreateFromString("0123456789abcdef".AsSpan()), ActivityTraceFlags.Recorded);
+            Assert.Equal("00-0123456789abcdef0123456789abcdef-0123456789abcdef-01", a.ParentId);
+
+            Activity b = new Activity("ParentFlagsB");
+            b.SetIdFormat(ActivityIdFormat.W3C);
+            b.SetParentId(ActivityTraceId.CreateFromString("0123456789abcdef0123456789abcdef".AsSpan()), ActivitySpanId.CreateFromString("0123456789abcdef".AsSpan()), ActivityTraceFlags.None);
+            b.ActivityTraceFlags = ActivityTraceFlags.Recorded; // Setting ActivityTraceFlags shouldn't affect the parent
+            Assert.Equal("00-0123456789abcdef0123456789abcdef-0123456789abcdef-00", b.ParentId);
+        }
```
Comment (noahfalk) [inline src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs:1710]: Does this case work? I am suspicious it won't yet
```C#
ActivityTraceId parentContext = new ActivityContext(
    ActivityTraceId.CreateFromString("0123456789abcdef0123456789abcdef".AsSpan()), 
    ActivitySpanId.CreateFromString("0123456789abcdef".AsSpan()), 
    ActivityTraceFlags.Recorded);
    using Activity? c = source.StartActivity(parentContext);
    Assert.Equal("00-0123456789abcdef0123456789abcdef-0123456789abcdef-00", c.ParentId);
```

**Review by noahfalk — APPROVED** [review]
I suspect there is a missing case, but if that one works already or you fix it everything else LGTM

File: src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs (hunk 1 above)
Comment (tarekgh) [reply src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs:1721]: You were right. I have fixed that case too. Thanks for the catch

**Pull request merged**